### PR TITLE
Add service detail page

### PR DESCRIPTION
**Pull request by westlywright**
Adds new component CreateEditView/Service which replaces both pages create/edit and detail with a single component that renders based on mode. As far as I can tell there is little downside to doing it this way as it reduces code duplication and contains all three views to a single component. 

rancher/dashboard#632

### CODE REVIEW

**Comment by vincent99** [comment]
I think this can be just tweaking a couple conditions in ResourceDetail to try the detail component if there is one, then try the edit, then YAML?  Then it would work for all types, without adding 2 one-line files per type.

File: edit/service.vue [hunk 1]
```diff
@@ -107,14 +128,27 @@ export default {
...
+      <template v-slot:created>
+        <LiveDate
+          :value="value.metadata.creationTimestamp"
+          :add-suffix="true"
+        />
+      </template>
+    </DetailTop>
     <form>
       <NameNsDescription
+        v-if="isCreate"
```
Comment (vincent99) [inline edit/service.vue:151]: !isView?  description should be editable

File: edit/service.vue [hunk 2]
```diff
@@ -107,14 +128,27 @@ export default {
 
       return false;
     }
-  }
+  },
+
+  updateExternalName(e) {
+    this.$set(this.value.spec, 'externalName', e);
+  },
```
Comment (vincent99) [inline edit/service.vue:135]: Nothing uses this?

File: components/ResourceYaml.vue [hunk 3]
```diff
@@ -256,13 +256,29 @@ export default {
       @onReady="onReady"
       @onChanges="onChanges"
     />
-    <Footer v-if="!isView" :mode="mode" :errors="errors" @save="save" @done="done">
```
Comment (vincent99) [inline components/ResourceYaml.vue:259]: Doesn't this make view always show a cancel button?

File: components/ResourceDetail/index.vue [hunk 4]
```diff
@@ -67,7 +68,9 @@ export async function defaultAsyncData(ctx, resource) {
 
   const hasCustomDetail = store.getters['type-map/hasCustomDetail'](resource);
   const hasCustomEdit = store.getters['type-map/hasCustomEdit'](resource);
-  const asYamlInit = (route.query[AS_YAML] === _FLAGGED) || (realMode === _VIEW && !hasCustomDetail) || (realMode !== _VIEW && !hasCustomEdit);
+  const asYamlInit = (hasIn(route.query, AS_YAML) && (route.query[AS_YAML] === _FLAGGED)) ||
```
Comment (vincent99) [inline components/ResourceDetail/index.vue:71]: What is the `hasIn` accomplishing here?

File: components/ResourceYaml.vue (hunk 3 above)
Comment (westlywright) [reply components/ResourceYaml.vue:259]: Yes, I found it odd we display the buttons everywhere but on view, but I am going to roll this back. It should be changed everywhere at once.

File: components/ResourceDetail/index.vue (hunk 4 above)
Comment (westlywright) [reply components/ResourceDetail/index.vue:71]: Its not needed. I'll drop it.

**Pull request merged**